contracts/NodeDelegator.sol

```diff
@@ -314,6 +388,169 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
+        uint256[] memory shares = new uint256[](1);
+        shares[0] = strategyShares;
+
+        // request the withdrawal of the LSTs from EigenLayer
+        IDelegationManager.QueuedWithdrawalParams[] memory requests = new IDelegationManager.QueuedWithdrawalParams[](1);
+        requests[0] = IDelegationManager.QueuedWithdrawalParams(strategies, shares, address(this));
+        address delegationManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_DELEGATION_MANAGER);
+
+        // request the withdrawal from EigenLayer.
+        // Will emit the Withdrawal event from EigenLayer's DelegationManager which is needed for the claim.
+        withdrawalRoot = IDelegationManager(delegationManagerAddress).queueWithdrawals(requests)[0];
+    }
+
+    /// @notice Claims the previously requested withdrawal from EigenLayer's underlying strategy.
+    /// Transfers the withdrawn assets to the staker that requested the withdrawal.
+    /// Is only callable by the `LRTDepositPool` contract.
+    /// @param withdrawal the `withdrawal` data emitted in the `WithdrawalQueued` event from EigenLayer's
+    /// `DelegationManager` contract when `requestWithdrawal` was called on the `LRTDepositPool` contract by the staker.
+    /// @param staker the address of the staker requesting the withdrawal
+    /// @return asset address of the liquid staking tokens (LST) that were claimed.
+    /// @return assets the amount of LSTs received from the withdrawal
+    function claimWithdrawal(
+        IDelegationManager.Withdrawal calldata withdrawal,
+        address staker
+    )
+        external
+        onlyDepositPool
+        whenNotPaused
+        returns (address asset, uint256 assets)
+    {
+        // Make sure the staker requested this withdrawal
+        // and the withdrawal was not manipulated
+        bytes32 withdrawalRoot = _calculateWithdrawalRoot(withdrawal);
+        if (withdrawalRequests[withdrawalRoot] != staker) {
+            revert StakersWithdrawalNotFound();
+        }
+
+        IERC20[] memory tokens = new IERC20[](1);
+        tokens[0] = IStrategy(withdrawal.strategies[0]).underlyingToken();
+        asset = address(tokens[0]);
+
+        // There can be assets sitting in this NodeDelegator contract so we need to account for those.
+        uint256 assetsBefore = IERC20(asset).balanceOf(address(this));
+
+        // Claim the previously requested withdrawal from EigenLayer
+        address delegationManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_DELEGATION_MANAGER);
+        // the 3rd middlewareTimesIndexes param is not used in the EigenLayer M2 contracts
+        IDelegationManager(delegationManagerAddress).completeQueuedWithdrawal(withdrawal, tokens, 0, true);
+
+        // Calculate the amount of assets returned from the withdrawal
+        assets = IERC20(asset).balanceOf(address(this)) - assetsBefore;
+        if (assets > 0) {
```

Comment:
Under which scenario will the assets be zero?
Gas wise if this check is necessary `!=` is always more gas efficient than `>`.

Reply:
It's an unlikely scenario where the user is redeeming a dust amount that gets rounded down to zero.